src/common/components/select-repository-list/index.js

```diff
@@ -96,41 +100,21 @@ export class SelectRepositoryListComponent extends Component {
     this.props.dispatch(fetchUserRepositories(pageNumber));
   }
 
-  filterEnabledRepos(repositories) {
-    const { success, snaps } = this.props.snaps;
-
-    if (success && snaps.length) {
-      for (let i = repositories.length; i--;) {
-        for (let j = snaps.length; j--;) {
-          const enabledRepo = snaps[j].git_repository_url;
-
-          if (enabledRepo === repositories[i].url) {
-            repositories[i].enabled = true;
```

Comment:
This was having a side effect of modifying redux state (without triggering any action).

Values passed from store as props are still references to objects in the store, so any modifications to them (or adding new properly) would appear in the store.